internal/controller/status.go

```diff
@@ -63,6 +75,83 @@ func podIsFailed(pod *v1.Pod) bool {
 	return false
 }
 
-func isIdle() bool {
+type MetricsSummary struct {
+	CPUMilicores int64
+	MemoryBytes  int64
+}
+
+type PodMetrics struct {
+	Kind       string            `json:"kind"`
+	APIVersion string            `json:"apiVersion"`
+	Metadata   metav1.ObjectMeta `json:"metadata"`
+
+	// The following fields define time interval from which metrics were
+	// collected in the following format [Timestamp-Window, Timestamp].
+	Timestamp metav1.Time     `json:"timestamp"`
+	Window    metav1.Duration `json:"window"`
+
+	// Metrics for all containers are collected within the same time window.
+	Containers []ContainerMetrics `json:"containers"`
+}
+
+type ContainerMetrics struct {
+	// Container name corresponding to the one from v1.Pod.Spec.Containers.
+	Name string `json:"name"`
+	// The memory usage is the memory working set.
+	Usage v1.ResourceList `json:"usage"`
+}
```

Comment:
@sgainst see https://github.com/SwissDataScienceCenter/amalthea/pull/742. I am sorry for being a stickler but I really wanted to:
- not depend on the concrete client but the interface
- not worry about extracting/parsing the response from the client - you can already call `Cpu()` on the result and get what you want
- not worry about parsing/converting the resource.Quatity objects - they have methods for this purpose already
- since we get the usage by container AND we have a fixed name for the session container which is always the same we can get the CPU usage only for the session container - this way we dont worry about adding up the usage and we also get a much more precise picture of what is going on

That PR I submitted targets this branch.

Reply:
No need to be sorry, it's a very good improvement ! Thanks !